compiler/luci/service/src/Nodes/CircleReshape.test.cpp

```diff
@@ -197,3 +197,39 @@ TEST(ShapeRuleTest, reshape_by_newShape)
   ASSERT_EQ(2, output_shape.dim(0).value());
   ASSERT_EQ(12, output_shape.dim(1).value());
 }
+
+TEST(ShapeRuleTest, reshape_by_newShape_dynamic)
+{
+  auto g = loco::make_graph();
+  auto node_reshape = g->nodes()->create<luci::CircleReshape>();
+  auto tensor_input = g->nodes()->create<luci::CircleInput>();
+  auto target_shape = g->nodes()->create<luci::CircleInput>();
+
+  tensor_input->dtype(loco::DataType::S32);
+  tensor_input->shape({2, 3, 4});
+  tensor_input->shape_status(luci::ShapeStatus::VALID);
+
+  target_shape->dtype(loco::DataType::S32);
+  target_shape->rank(1);
+  target_shape->shape_status(luci::ShapeStatus::VALID);
+
+  node_reshape->tensor(tensor_input);
+  node_reshape->shape(target_shape);
+
+  // reshape to {dynamic, 4, dynamic}
+  node_reshape->newShape()->rank(3);
+  node_reshape->newShape()->dim(0) = -1;
+  node_reshape->newShape()->dim(1) = 2;
```

Comment:
Now you changed `4` to `2`. What's the rationale behind this?

Reply:
the idea was to set data easier to understand (especially after shape specialization)